### PR TITLE
[DX] [Security] Renamed Token#getKey() to getSecret()

### PR DESCRIPTION
**Pull request by wouterj**
There are 2 very vague parameter names in the authentication process: `$providerKey` and `$key`. Some tokens/providers have the first one, some tokens/providers the second one and some both. An overview:

| Token | `providerKey` | `key` |
| --- | --- | --- |
| `AnonymousToken` | - | yes |
| `PreAuth...Token` | yes | - |
| `RememberMeToken` | yes | yes |
| `UsernamePasswordToken` | yes | - |

Both names are extremely general and their PHPdocs contains pure no-shit-sherlock-descriptions :squirrel: (like "The key."). This made me and @iltar think it's just an inconsistency and they have the same meaning.
...until we dived deeper into the code and came to the conclusion that `$key` has a Security task (while `$providerKey` doesn't really). If it takes people connected to Symfony internals 30+ minutes to find this out, it should be considered for an improvement imo.

So here is our suggestion: **Rename `$key` to `$secret`**. This explains much better what the value of the string has to be (for instance, it's important that the string is not easily guessable and cannot be found out, according to the Spring docs). It also explains the usage better (it's used as a replacement for credentials and to hash the RememberMeToken).

**Tl;dr**: `$key` and `$providerKey` are too general names, let's improve DX by renaming them. This PR tackles `$key` by renaming it to `$secret`.

| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | yes |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

_My excuse for the completely unrelated branch name_

### CODE REVIEW

**Comment by stof** [comment]
you need to bump the min requirement of ``symfony/security-core`` in ``symfony/security-http``

**Comment by WouterJ** [comment]
Thanks @stof, done :)

**Comment by stof** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @WouterJ.
